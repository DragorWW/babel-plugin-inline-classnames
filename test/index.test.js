/* eslint-env jest */
/* eslint no-console: 0 */

const babel = require('babel-core');
const fs = require('fs');

const babelOptions = {
	plugins: ['syntax-jsx', __dirname + '/../index.js'],
};

[
	'basic',
	'custom-import-name',
	'other-classnames',
	'string-literal',
	'object-expression',
	'falsy-values',
	'classnames-examples',
].forEach(test => {
	it(test, () => {
		const source = fs.readFileSync(__dirname + '/fixtures/' + test + '.js');
		const { code } = babel.transform(source, babelOptions);

		expect(code).toMatchSnapshot();
	});
});
