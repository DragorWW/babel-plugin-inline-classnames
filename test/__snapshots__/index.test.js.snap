// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"
import styles from './styles.css';

styles.foo + ' ' + styles.bar;

<div className={styles.foo + ' ' + styles.bar} />;"
`;

exports[`classnames-examples 1`] = `
"

'foo bar';
'foo bar';
'foo-bar';
'';
'foo bar';
'foo bar';
'foo bar baz quux';
'bar ' + 1;"
`;

exports[`custom-import-name 1`] = `
"
import styles from '../styles.css';

styles.foo + ' ' + styles.bar;

<div className={styles.foo + ' ' + styles.bar} />;"
`;

exports[`falsy-values 1`] = `
"

'';
'foo ' + 1;"
`;

exports[`object-expression 1`] = `
"
import styles from './styles.css';

'foo bar';
'foo-bar';
'foo bar';
'foo bar';
'foo ham';
styles.foo + ' ' + styles.bar;"
`;

exports[`other-classnames 1`] = `
"import styles from './styles.css';

function classnames() {}

classnames(styles.foo, styles.bar);

<div className={classnames(styles.foo, styles.bar)} />;"
`;

exports[`scoped 1`] = `
"

function scope1() {
	function scope2() {
		return 'foo bar';
	}
}"
`;

exports[`string-literal 1`] = `
"
import styles from './styles.css';

'global ' + (styles.foo + ' ' + styles.bar);
'foo bar ham';"
`;
